io/io/src/TFile.cxx

```diff
@@ -4240,6 +4240,23 @@ TFile *TFile::Open(const char *url, Option_t *options, const char *ftitle,
    return f;
 }
 
+////////////////////////////////////////////////////////////////////////////////
+/// \brief Call TFile::Open and return a std::unique_ptr.
+///
+/// \param[in] url Full path to the (remote) file.
+/// \param[in] options Options to pass when opening the file.
+/// \param[in] ftitle File title.
+/// \param[in] compress ROOT compression settings.
+/// \param[in] netopt Argument specific to TNetFile.
+///
+/// This function calls TFile::Open passing the input arguments as they are. The
+/// returned file is wrapped in a unique_ptr.
+std::unique_ptr<TFile>
+TFile::MakeUnique(const char *url, Option_t *options, const char *ftitle, Int_t compress, Int_t netopt)
+{
+   return std::unique_ptr<TFile>{TFile::Open(url, options, ftitle, compress, netopt)};
```

Comment:
@pcanal would it make sense to de-register this file from the cleanups?

Reply:
> The fact that make_unique<TFile> does something else is the fault of the public TFile ctor ;-)

No it is not :).  [It might be the 'fault' of the design pattern but one is an explicit creation of an explicit type, the other is calling a factory function].

> TFile::MakeUnique(): follows a known pattern!

Sorta.  But why would:
```
std::make_unique<TFile> file(finename);
// and
auto file = TFile::MakeUnique(finename);
```
behave differently?  How easy will  it be for user to confuse both? (Converse how hard will it be to explain it).

> would it make sense to de-register this file from the cleanups?

No it would not.  The list of cleanups is the list of object that needs to be informed when another object is deleted (eg. the TTree object).
In addition it is not added directly to the list of clean-ups. Instead it is the list of "opened" files .. which is used, for example, by the `TBrowser` to show the list opened files,  so it probably does not.

> `TFile::OpenUniquePtr()`: we're not opening a `unique_ptr`

So we need something that says: `TFile::Open_ReturningAUniquePtr` .... which I guess we could spell :):) :
```
std:unique_ptr<TFile>{ TFile::Open( ... ) };
```

That said, I agree with your points against `OpenUnique[Ptr]` so maybe `TFile::OpenIntoUniquePtr` 

Alternatively (ideally?) `TFile::Open` would return a 'sorta-unique_ptr` that would automatically convert to `TFile*` and a `std::unique_ptr`, so that it is backward compatible:
usage example:
```
TFile *fileptr = TFile::Open(filename); // still work thanks to `TFile*` conversion.
std:unique_ptr<TFile> unique{ TFile::Open( ... ) }; // still work thanks to `std::unique_ptr` conversion.
...

auto file = TFile::Open(filename);
...
function_taking_tfile_start(file);  // still work thanks to `TFile*` conversion.
```

[Note: 'sorta-unique-ptr` is anyway probably 'truer' semantically since the pointer is actually shared with the list of file and can thus be deleted, for example, from the graphical interface]

Reply:
Ooooh that's a nice idea! It allows us to document that the returned pointer needs ownership handling by the user! Let's discuss this on Monday - this might be a big one :-)